src/cpp/src/vision_encoder.cpp

```diff
@@ -0,0 +1,217 @@
+                clip_image_u8& patch = images.back().back();
+                patch.nx = grid_x;
+                patch.ny = grid_y;
+                patch.buf.resize(3 * patch.nx * patch.ny);
+                for (int y = patches_i; y < patches_i + grid_y; ++y) {
+                    for (int x = patches_j; x < patches_j + grid_x; ++x) {
+                        const int i = 3 * (y * refine_image.nx + x);
+                        const int j = 3 * ((y - patches_i) * patch.nx + (x - patches_j));
+                        patch.buf[j] = refine_image.buf[i];
+                        patch.buf[j + 1] = refine_image.buf[i + 1];
+                        patch.buf[j + 2] = refine_image.buf[i + 2];
+                    }
+                }
+            }
+        }
+    }
+
+    return images;
+}
+
+EncodedImage llava_image_embed_make_with_bytes_slice(clip_ctx& ctx_clip, const ov::Tensor& img, ov::InferRequest& encoder, int max_slice_nums, int scale_resolution, size_t patch_size, bool never_split) {
+    clip_image_u8 source{
+        int(img.get_shape().at(3)),
+        int(img.get_shape().at(2)),
+        {img.data<uint8_t>(), img.data<uint8_t>() + img.get_size()}
+    };
+    std::vector<std::vector<clip_image_u8>> imgs = ::slice_image(source, max_slice_nums, scale_resolution, patch_size, never_split);
+    std::vector<std::vector<ov::Tensor>> results;
+    std::vector<std::vector<HeightWidth>> sizes;
+
+    // std::vector<clip_image_f32*> img_res_v; // format VectN x H x W x RGB (N x 336 x 336 x 3), so interleaved RGB - different to the python implementation which is N x 3 x 336 x 336
+    std::vector<std::vector<clip_image_f32>> preprocessed{imgs.size()};
+    std::transform(imgs.begin(), imgs.end(), preprocessed.begin(), [&ctx_clip](const std::vector<clip_image_u8>& row) {
+        std::vector<clip_image_f32> processed_row{row.size()};
+        std::transform(row.begin(), row.end(), processed_row.begin(), [&ctx_clip](const clip_image_u8& raw) {
+            return clip_image_preprocess(ctx_clip, raw);
+        });
+        return processed_row;
+    });
+
+    const clip_image_f32& resized_preprocessed = preprocessed.at(0).at(0);
+    ov::Tensor input_tensor{ov::element::f32, {1, 3, size_t(resized_preprocessed.ny), size_t(resized_preprocessed.nx)}, (void*)(resized_preprocessed.buf.data())};
+    encoder.set_input_tensor(input_tensor);
+    encoder.infer();
+    ov::Tensor output_tensor = encoder.get_output_tensor();
+    ov::Tensor resized_source{output_tensor.get_element_type(), output_tensor.get_shape()};
+    output_tensor.copy_to(resized_source);
+    HeightWidth resized_source_size{resized_preprocessed.ny / patch_size, resized_preprocessed.nx / patch_size};
+
+    HeightWidth size{
+        size_t(preprocessed.at(1).at(0).ny),
+        size_t(preprocessed.at(1).at(0).nx)
+    };
+    ov::Tensor batched{ov::element::f32, {(preprocessed.size() - 1) * preprocessed.at(1).size(), 3, size.height, size.width}};
```

Comment:
`preprocessed.at(1)` causes vector index out of bounds.
Should it be `preprocessed.at(0)`?

Reply:
No. There's `preprocessed.size() - 1` instead of `preprocessed.size()` for the same reason. The first image is a resized source image and everything else are slices

Reply:
I think the problem is that small enough images aren't sliced. I don't handle it and your image is like that.